app/controllers/carts_controller.rb

```diff
@@ -1,4 +1,6 @@
 class CartsController < ApplicationController
+  include CurrentCart
+  load_and_authorize_resource
```

Comment:
types of cancancan methods, load and authoriize resource, authorize resource etc differences